### ISSUE TITLE
[Feature Request] use location.href instead of location.replace

### ISSUE THREAD

**Issue by GonziHere** [proginosko/LeechBlockNG#236]
Hi, thanks for absolutely stellar productivity tool.

My only issue is that when I open some random link that redirects me to youtube (for example) and I'm not supposed to be there, I'll loose the link altogether. This forces me to override, which might lead to actually watching it. 

I use the "Enter the fully specified URL of the page to show instead of these blocked sites" feature ( https://i.imgflip.com/5zv51d.jpg 🤣 ), but I would still like to maintain the original url, so that I can easily get back to in while on break for example.

You might have a different solution in mind - I'm fine with anything, really, as long as I don't loose the link itself :). From the top of my head:

1. maybe add url to history so that I can go back outside of time limits.
2. maybe add some overlay to the redirect that can be "refreshed".
3. maybe expand options for "delayed" page to allow some custom content.
4. FANCY: maybe add the option to add the link to "visit later" folder that the user can create in the browser? 
5. ???

**Comment by proginosko** [comment]
Thanks for the suggestion. I assume you want to retain the original URL, rather than the URL to which it redirects, correct?

**Comment by GonziHere** [comment]
Yes. My first thought was that I can "refresh" the original link with a simple go back action.

**Comment by proginosko** [comment]
Would you be able to do that if LeechBlock weren't blocking the site? I'm just trying to ascertain what kind of redirect is in view.

**Comment by GonziHere** [comment]
I use leechblock for youtube with a custom url for "**page to show instead of blocked sites**". This means that when I search for something and the result is on youtube, I will loose the youtube link, since my browser history will have something like 

1. google/search/how-to-x
2. https://i.imgflip.com/5zv51d.jpg

Say that it's actually related to the work (which happens a lot), I'll click override and... I cannot refresh (will still be the override url) and I cannot go back, since it would return me to the google/search/how-to-x. This is especially annoying if I open 4 howto links from google, and 1 random result will be from youtube.

This issue is related to the "page to show instead of blocked sites". I believe that the other methods (blur certainly and delay page most likely) don't loose the url that I was trying to visit.

I want to have in my history this:

1. google/search/how-to-x 
2. youtube/v=howtox
3. https://i.imgflip.com/5zv51d.jpg

So that I can go back to it (when the restrictions are lifted).

For the record, my current workaround is to use https://i.imgflip.com/5zv51d.jpg?from=$U , use override and then modify the url, but that's quite hacky.

**Comment by proginosko** [comment]
The tricky thing here is that LB is designed to intercept the page as soon as the URL is activated in the browser, before the page is even loaded, and before the URL enters the browser history. But there may be a way to add it to the history anyway. I'll look into it.

**Comment by GonziHere** [comment]
I see, thanks for the info. If you manage, then great, if you don't, thanks anyway :).

btw, maybe some of the alternative solutions to this problem might be worth looking into (as in customizable waiting page, or waiting overlay over custom url and so on and so forth). History is just the first solution that has come to my mind, since I was hoping for a simple replace of the js call (as in title).
